File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/uncomment/direct/Java/C/s081988820_commented.c
#include <stdio.h>

int main() {
    int H, W;
    int c[10][10];
    int min[10];
    int tc, i, j, h, w, A, ans = 0;

    scanf("%d %d", &H, &W);

    for (i = 0; i < 10; i++) {
        for (j = 0; j < 10; j++) {
            scanf("%d", &c[i][j]);
        }
    }

    for (i = 0; i < 10; i++) {
        min[i] = c[i][1];
    }

    for (tc = 0; tc < 10; tc++) {
        for (i = 0; i < 10; i++) {
            for (j = 0; j < 10; j++) {
                if (min[i] > c[i][j] + min[j]) {
                    min[i] = c[i][j] + min[j];
                }
            }
        }
    }

    ans = 0;
    for (h = 0; h < H; h++) {
        for (w = 0; w < W; w++) {
            scanf("%d", &A);
            if (A >= 0) {
                ans += min[A];
            }
        }
    }

    printf("%d\n", ans);
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/uncomment/direct/Java/Go/s081988820_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)
	var nums []int
	for sc.Scan() {
		num, _ := strconv.Atoi(sc.Text())
		nums = append(nums, num)
	}
	H := nums[0]
	W := nums[1]
	idx := 2
	c := make([][]int, 10)
	for i := range c {
		c[i] = make([]int, 10)
		for j := range c[i] {
			c[i][j] = nums[idx]
			idx++
		}
	}
	min := make([]int, 10)
	for i := 0; i < 10; i++ {
		min[i] = c[i][1]
	}
	for tc := 0; tc < 10; tc++ {
		for i := 0; i < 10; i++ {
			for j := 0; j < 10; j++ {
				if min[i] > c[i][j]+min[j] {
					min[i] = c[i][j] + min[j]
				}
			}
		}
	}
	ans := 0
	for ; idx < len(nums); idx++ {
		A := nums[idx]
		if A >= 0 {
			ans += min[A]
		}
	}
	fmt.Println(ans)
}

// End of Code
